usr.bin/look/look.c

```diff
@@ -132,21 +138,35 @@ main(int argc, char *argv[])
 
 	match = 1;
 
-	do {
-		if ((fd = open(file, O_RDONLY, 0)) < 0 || fstat(fd, &sb))
+#ifndef WITHOUT_CAPSICUM
+	cap_rights_init(&rights, CAP_MMAP_R, CAP_READ, CAP_FSTAT);
+#endif
+	int fds[argc > 1 ? argc - 1 : argc];
+	for (size_t idx = 0; idx < sizeof fds / sizeof fds[0]; file = argv[idx++]) {
+		if ((fds[idx] = open(file, O_RDONLY, 0)) < 0)
+			err(2, "%s", file);
+#ifndef WITHOUT_CAPSICUM
+		if (cap_rights_limit(fds[idx], &rights) < 0)
+			err(2, "unable to limit rights for %s", file);
+	}
+	if (cap_enter() < 0)
+		err(EXIT_FAILURE, "failed to enter capability mode");
+	for (size_t idx = 0; idx < sizeof fds / sizeof fds[0]; file = argv[idx++]) {
```

Comment:
We use sizeof().
You can also use nitems.

Reply:
This if is under WITHOUT_CAPSICUM?
Have you tried to build this with this flag defined?

Reply:
Yes, and it compiled and worked as expected. But I'll double check shortly.

Reply:
And you're absolutely right regarding `sizeof()`, we have it clearly stated in style(9), my apologies.